src/XML/AbstractSignedXMLElement.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\XML;
+
+use DOMElement;
+use SimpleSAML\XMLSecurity\XML\ds\Signature;
+
+/**
+ * Abstract class to be implemented by all signed classes
+ *
+ * @package simplesamlphp/xml-security
+ */
+abstract class AbstractSignedXMLElement implements SignedElementInterface
+{
+    use SignedElementTrait;
+
+    /**
+     * The signed DOM structure.
+     *
+     * @var \DOMElement
+     */
+    protected DOMElement $structure;
+
+    /**
+     * The unsigned elelement.
+     *
+     * @var \SimpleSAML\XMLSecurity\XML\SignableElementInterface
+     */
+    protected SignableElementInterface $element;
+
+
+    /**
+     * Create/parse an alg:SigningMethod element.
+     *
+     * @param \DOMElement $xml
+     * @param \SimpleSAML\XMLSecurity\XML\SignableElementInterface $elt
+     * @param \SimpleSAML\XMLSecurity\XML\ds\Signature $signature
+     */
+    protected function __construct(DOMElement $xml, SignableElementInterface $elt, Signature $signature)
```

Comment:
As I mentioned in another comment, I would remove the actual element from the constructor, so that it is only built after verifying the signature.